### PR TITLE
v4.1 upgrade guide

### PR DESCRIPTION
**Pull request by kennyadsl**
## Summary

Target merge date: ~Tuesday, 27/06/23~ Thursday 30/06/23

Adds Upgrade Guides for v4.1

## Checklist

- [ ] I have followed the [Diátaxis](https://diataxis.fr/) framework in my PR.
- [ ] I have verified that the preview environment works correctly.

### CODE REVIEW

**Review by jarednorman — APPROVED** [review]

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @kennyadsl; everything is super clear. I think we're now seeing the benefits of the new release cadence. As you said, it's a tiny release but still has stuff worth highlighting. Delivering with small chunks makes changes more digestible.

File: docs/upgrading-solidus/v4.1.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+## <PRLink description="Allow changing the order recalculator" number="5110" />
+
+The `order.recalculate` and `order.recalculator` are introduced as aliases to the now deprecated
+`order.update` and `order.updater` in order to clarify the intention of these methods.
+
+This change also adds a preference to change the order recalculator (also known as order updater).
+The order recalculator, a very central piece of Solidus' infrastructure, carries a bit of technical debt.
+Unfortunately, it's extremely hard to change because every line of this class might have been overwritten
+by implementors.
```
Comment (elia) [inline docs/upgrading-solidus/v4.1.mdx:53]: The tone here is a bit gloomy, but I guess anyone that ever worked with the _recalculator_ will know 😅 
(just a consideration, no change needed)

**Review by elia — APPROVED** [review]

File: docs/upgrading-solidus/v4.1.mdx (hunk 1 above)
Comment (kennyadsl) [reply docs/upgrading-solidus/v4.1.mdx:53]: You are right, I can omit that part: it is not relevant in the upgrade phase.

Comment (kennyadsl) [reply docs/upgrading-solidus/v4.1.mdx]: Done!

**Pull request merged**